src/components/application_manager/test/commands/mobile/on_hash_change_notification_test.cc

```diff
@@ -0,0 +1,114 @@
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "application_manager/commands/mobile/on_hash_change_notification.h"
+#include "command_impl.h"
+#include "commands/commands_test.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/smart_object_keys.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+namespace strings = application_manager::strings;
+
+using application_manager::commands::mobile::OnHashChangeNotification;
+using application_manager::MockMessageHelper;
+using application_manager::commands::CommandImpl;
+using testing::Return;
+using testing::ReturnRef;
+using testing::_;
+
+class OnHashChangeNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  OnHashChangeNotificationTest()
+      : message_helper_(*MockMessageHelper::message_helper_mock()) {}
+  MockMessageHelper& message_helper_;
+};
+
+TEST_F(OnHashChangeNotificationTest, Run_ValidApp_SUCCESS) {
+  const uint32_t kConnectionKey = 1u;
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+
+  SharedPtr<OnHashChangeNotification> command =
+      CreateCommand<OnHashChangeNotification>(msg);
+
+  std::string return_string;
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  EXPECT_CALL(*mock_app, curHash()).WillOnce(ReturnRef(return_string));
+  EXPECT_CALL(message_helper_, PrintSmartObject(_)).WillOnce(Return(false));
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(msg, _));
+
+  command->Run();
+
+  ASSERT_EQ(application_manager::MessageType::kNotification,
+            (*msg)[strings::params][strings::message_type].asInt());
+  ASSERT_EQ(CommandImpl::mobile_protocol_type_,
+            (*msg)[strings::params][strings::protocol_type].asInt());
+  ASSERT_EQ(CommandImpl::protocol_version_,
+            (*msg)[strings::params][strings::protocol_version].asInt());
+}
+
+TEST_F(OnHashChangeNotificationTest, Run_InvalidApp_SUCCESS) {
+  const uint32_t kConnectionKey = 1u;
```

Comment:
@pvvasilev, better to look here: 
https://adc.luxoft.com/confluence/pages/viewpage.action?pageId=267764838
...
4. Best practices in unit test writing :
...
Best way to name any test is using template : [TestedMethodName]_[Scenario]_[ExpectedBehavior]
...
ExpectedBehavior - what are we wait in the behavior of method, e.g. if we wait returning of 7 "Sum_TwoAndFive_Seven"
...
For me SUCCESS means that normal (positive) sequence is executed which is not the case with invalid data. For me "success" doesn't mean that function/method was executed without crash or other strange behaviour - we are talking here from high-level point of view - is it expected to succeed with wrong data? I think it is expected to not, so it is expected high-level operation to fail in this case. And if operation fails - it is Okay (expected). It will be problematic if succeeds with invalid data (in this case UT must report it as failed/error).